Cannot find module '@farmfe/core/config' or its corresponding type declarations

### Steps to reproduce

1. `yarn create farm`
2. `yarn add typescript --dev`
3. `yarn tsc --target esnext --noEmit farm.config.ts`

### What is expected?

No type errors. 

### What is actually happening?

```
farm.config.ts:1:34 - error TS2307: Cannot find module '@farmfe/core/config' or its corresponding type declarations.

1 import { defineFarmConfig } from '@farmfe/core/config';
```

### System Info

```Shell
OS: macOS 13.3.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 153.55 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.18.0 - ~/.asdf/installs/nodejs/16.18.0/bin/node
    Yarn: 1.22.17 - ~/.asdf/shims/yarn
    npm: 8.19.2 - ~/.asdf/plugins/nodejs/shims/npm
    Watchman: 2023.03.27.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 112.0.5615.121
    Firefox: 107.0.1
    Safari: 16.4
  npmPackages:
    @farmfe/core: latest => 0.6.3
```


### Any additional comments?

I notice that there is a specific type definition for the `./config` export map, but it doesn't seem to matter. 

Importing like this works fine (but of course less elegant):
```
import { defineFarmConfig } from "@farmfe/core/dist/config";
```

Maybe consider adding a declare section to the root type definition (in `dist/index.d.ts`) or hoist contents of dist folder? Not sure what the correct approach is.

--- a/.changeset/lazy-crabs-confess.md
+++ b/.changeset/lazy-crabs-confess.md
@@ -1,0 +1,5 @@
+---
+'create-farm': patch
+---
+
+Remove the use of defineFarmConfig from the templates' farm.config.ts

--- a/packages/create-farm/templates/react/farm.config.ts
+++ b/packages/create-farm/templates/react/farm.config.ts
@@ -1,13 +1,12 @@
-import { defineFarmConfig } from '@farmfe/core/config';
+import type { UserConfig } from '@farmfe/core';
 
-export default defineFarmConfig({
+export default <UserConfig>{
   compilation: {
     input: {
       index: './index.html',
     },
     resolve: {
       symlinks: true,
-      mainFields: ['module', 'main', 'customMain'],
     },
     define: {
       BTN: 'Click me',
@@ -20,4 +19,4 @@ export default defineFarmConfig({
     hmr: true,
   },
   plugins: ['@farmfe/plugin-react'],
-});
+};

--- a/packages/create-farm/templates/react/package.json
+++ b/packages/create-farm/templates/react/package.json
@@ -12,7 +12,8 @@
     "@farmfe/plugin-react": "0.1.2",
     "@types/react": "18",
     "@types/react-dom": "18",
-    "react-refresh": "^0.14.0"
+    "react-refresh": "^0.14.0",
+    "typescript": "^4.5.2"
   },
   "scripts": {
     "start": "farm start",

--- a/packages/create-farm/templates/react/src/index.tsx
+++ b/packages/create-farm/templates/react/src/index.tsx
@@ -3,6 +3,8 @@ import { createRoot } from 'react-dom/client';
 import { Main } from './main';
 
 const container = document.querySelector('#root');
-const root = createRoot(container);
 
-root.render(<Main />);
+if (container) {
+  const root = createRoot(container);
+  root.render(<Main />);
+}

--- a/packages/create-farm/templates/react/tsconfig.json
+++ b/packages/create-farm/templates/react/tsconfig.json
@@ -1,0 +1,12 @@
+{
+  "compilerOptions": {
+    "lib": ["DOM", "ES2017"],
+    "noEmit": true,
+    "esModuleInterop": true,
+    "jsx": "react",
+    "skipDefaultLibCheck": true,
+    "skipLibCheck": true
+  },
+  "include": ["src"],
+  "exclude": ["node_modules"]
+}

--- a/packages/create-farm/templates/vue/.eslintrc.json
+++ b/packages/create-farm/templates/vue/.eslintrc.json
@@ -1,8 +1,0 @@
-{
-  "root": true,
-  "extends": "../../.eslintrc.base.json",
-  "parserOptions": {
-    "project": ["./examples/vue/tsconfig.json"]
-  },
-  "rules": {}
-}

--- a/packages/create-farm/templates/vue/CHANGELOG.md
+++ b/packages/create-farm/templates/vue/CHANGELOG.md
@@ -1,7 +1,0 @@
-# @farmfe-examples/vue
-
-## 0.0.1
-
-### Patch Changes
-
-- 9060d9c: Add the auto installing and the check of the repeat project name

--- a/packages/create-farm/templates/vue/farm.config.ts
+++ b/packages/create-farm/templates/vue/farm.config.ts
@@ -1,8 +1,7 @@
-// change to @farmfe/core/config when resolve support conditional exports
-import { defineFarmConfig } from '@farmfe/core/config';
+import type { UserConfig } from '@farmfe/core';
 import farmJsPluginVue from '@farmfe/js-plugin-vue';
 
-export default defineFarmConfig({
+export default <UserConfig>{
   compilation: {
     input: {
       index: './index.html',
@@ -15,4 +14,4 @@ export default defineFarmConfig({
     },
   },
   plugins: [farmJsPluginVue()],
-});
+};

--- a/packages/create-farm/templates/vue/src/env.d.ts
+++ b/packages/create-farm/templates/vue/src/env.d.ts
@@ -1,2 +1,2 @@
-declare module "*.vue";
-declare module "*.svg";
+declare module '*.svg';
+declare module '*.png';

